include/lsst/modelfit/parameters/type_name.h

```diff
@@ -21,13 +22,13 @@
  * along with this program.  If not, see <https://www.gnu.org/licenses/>.
  */
 
-#ifndef PARAMETERS_TYPE_NAME_H
-#define PARAMETERS_TYPE_NAME_H
+#ifndef LSST_MODELFIT_PARAMETERS_TYPE_NAME_H
+#define LSST_MODELFIT_PARAMETERS_TYPE_NAME_H
 
 #include <string>
 #include <string_view>
 
-namespace parameters {
+namespace lsst::modelfit::parameters {
 
 // https://stackoverflow.com/questions/81870/is-it-possible-to-print-a-variables-type-in-standard-c/64490578#64490578
 template <typename T>
```

Comment:
Please add documentation comments to all this code. You might also consider [using a `::detail` namespace](https://developer.lsst.io/cpp/style.html#implementation-specific-globals-should-go-in-namespace-detail).

Reply:
There is a detail namespace already. `type_name` and `type_name_str` generally need to be called by implementing transforms now, so I think they ought not to be in the detail namespace.